mautic/mautic
# Add html custom field

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | N
| New feature/enhancement? (use the a.x branch)      | Y
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | Y 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR implements a new HTML custom type field.
It allows to store html data and use it in emails, dynamic content and view/edit it in Mautic just as any other custom field.



#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new custom field with type HTML for Contacts.
3. Create a new contact, add something like `<h1>Prague</h1>` to the field you've just created. Save the contact.
4. View that contact in Mautic. Make sure you see the `<h1>Prague</h1>` text in the details tray.
5. Go to to the audit log for that contact.
6. Make sure you see the `<h1>Prague</h1>` text in the audit log.
7. Create contact via API, send value `<h1>Prague</h1>` in your html field
8. Open the contact in Mautic. Make sure you see the `<h1>Prague</h1>` text.
9. Create a segment email. Insert a token in that email that is using the html field you created on step 3. LIke {contactfield=your_html_field}
10. Send an email to yourself.
11. You should see unescaped html code in the email.
